packages/grid/src/vaadin-grid-a11y-mixin.js

```diff
@@ -103,13 +103,39 @@ export const A11yMixin = (superClass) =>
     _a11yUpdateRowExpanded(row) {
       if (this.__isRowExpandable(row)) {
         row.setAttribute('aria-expanded', 'false');
+        const toggleCell = this.__a11yFindTreeToggleCell(row);
+        if (toggleCell) {
+          toggleCell.setAttribute('aria-expanded', 'false');
+        }
       } else if (this.__isRowCollapsible(row)) {
         row.setAttribute('aria-expanded', 'true');
+        const toggleCell = this.__a11yFindTreeToggleCell(row);
+        if (toggleCell) {
+          toggleCell.setAttribute('aria-expanded', 'true');
+        }
       } else {
         row.removeAttribute('aria-expanded');
+        const toggleCell = this.__a11yFindTreeToggleCell(row);
+        if (toggleCell) {
+          toggleCell.removeAttribute('aria-expanded');
+        }
       }
     }
 
+    /**
+     * Finds the cell containing the tree toggle element
+     * @param {!HTMLElement} row
+     * @return {HTMLElement|null}
+     * @private
+     */
+    __a11yFindTreeToggleCell(row) {
```

Comment:
Let's create this helper function in `vaadin-grid-helpers.js` to keep it separate from the class logic.

Reply:
Extracted the function.